src/engine/GroupBy.cpp

```diff
@@ -725,64 +728,239 @@ GroupBy::checkIfHashMapOptimizationPossible(
 
   // TODO<kcaliban> remove this as soon as we have a better implementation
   //                for multiple aggregates
-  // Only allow up to 5 aggregates for now (because of CallFixedSize)
+  // Only allow up to 5 aliases for now (because of CallFixedSize)
   if (aggregates.size() > 5) {
     return std::nullopt;
   }
 
-  for (auto& aggregate : aggregates) {
-    auto expr = aggregate._expression.getPimpl();
+  // Get pointers to all aggregate expressions and their parents
+  size_t numAggregates = 0;
+  std::vector<HashMapAliasInformation> aliasAggregates;
+  for (auto& aggregateAlias : aggregates) {
+    std::vector<HashMapAggregateInformation> aliasAggregateInfo;
+    auto expr = aggregateAlias._expression.getPimpl();
 
-    // Only allow AVG for now
-    if (dynamic_cast<const sparqlExpression::AvgExpression*>(expr) == nullptr)
+    if (!findAggregateMultiple(nullptr, expr, std::nullopt, aliasAggregateInfo))
```

Comment:
We have to return `std::nullopt` since we need to know if there are unsupported aggregates (anything other than `AVG` at the moment), so that we disable usage of the optimization.